src/System.Net.NetworkInformation/tests/FunctionalTests/NetworkAvailabilityChangedTests.cs

```diff
@@ -27,6 +27,20 @@ public void NetworkAvailabilityChanged_JustRemove_Success()
             NetworkChange.NetworkAvailabilityChanged -= _availabilityHandler;
         }
 
+        [Fact]
+        [ActiveIssue(33530, TestPlatforms.FreeBSD)]
+        public void NetworkAddressChanged_Add_DoesNotBlock()
+        {
+            // Register without unregistering.
+            // This should not block process exit. If it does, this test will pass
+            // but we would fail to exit test run at the end.
+            // We cannot test this via RemoteInvoke() as that call System.Exit()
```

Comment:
```suggestion
            // We cannot test this via RemoteInvoke() as that calls System.Exit()
```

Reply:
Indeed it even says
```
                // Use Exit rather than simply returning the exit code so that we forcibly shut down
                // the process even if there are foreground threads created by the operation that would
                // end up keeping the process alive potentially indefinitely.
```

Reply:
I know I added this explicitly a while back, but I can't remember now why.  Ideally there wouldn't be any foreground threads and this logic could go away.  Worth looking into subsequently.